src/wireviz/wv_dataclasses.py

```diff
@@ -0,0 +1,815 @@
+            )
+        return _hash
+
+    @property
+    def bom_qty(self) -> float:
+        if self.sum_amounts_in_bom:
+            if self.amount:
+                return self.qty.number * self.amount.number
+            else:
+                return self.qty.number
+        else:
+            return self.qty.number
+
+    def bom_amount(self) -> NumberAndUnit:
+        if self.sum_amounts_in_bom:
+            return NumberAndUnit(None, None)
+        else:
+            return self.amount
+
+    @property
+    def has_pn_info(self) -> bool:
+        return any([self.pn, self.manufacturer, self.mpn, self.supplier, self.spn])
+
+
+@dataclass
+class AdditionalComponent(Component):
+    qty_multiplier: Union[QtyMultiplierConnector, QtyMultiplierCable, int] = 1
+    _qty_multiplier_computed: Union[int, float] = 1
+    designators: Optional[str] = None  # used for components definedi in the
+    #                                    additional_bom_items section within another component
+    bgcolor: SingleColor = None  #       ^ same here
+    note: str = None
+
+    def __post_init__(self):
+        super().fill_partnumbers()
+        self.bgcolor = SingleColor(self.bgcolor)
+        self.qty = self.parse_number_and_unit(self.qty, None)
+        self.amount = self.parse_number_and_unit(self.amount, None)
+
+        if isinstance(self.qty_multiplier, float) or isinstance(
+            self.qty_multiplier, int
+        ):
+            pass
+        else:
+            self.qty_multiplier = self.qty_multiplier.upper()
+            if self.qty_multiplier in QtyMultiplierConnector.__members__.keys():
+                self.qty_multiplier = QtyMultiplierConnector[self.qty_multiplier]
+            elif self.qty_multiplier in QtyMultiplierCable.__members__.keys():
+                self.qty_multiplier = QtyMultiplierCable[self.qty_multiplier]
+            else:
+                raise Exception(f"Unknown qty multiplier: {self.qty_multiplier}")
+
+    @property
+    def additional_components(self):
+        # an additional component may not have further nested additional comonents
+        return []
+
+    @property
+    def bom_qty(self):
+        return self.qty.number * self._qty_multiplier_computed
+
+    @property
+    def description(self) -> str:
+        return f"{self.type}{', ' + self.subtype if self.subtype else ''}"
+
+
+@dataclass
+class GraphicalComponent(Component):  # abstract class, for future use
+    bgcolor: Optional[SingleColor] = None
+
+
+@dataclass
+class TopLevelGraphicalComponent(GraphicalComponent):  # abstract class
+    # component properties
+    designator: Designator = None
+    color: Optional[MultiColor] = None
+    image: Optional[Image] = None
+    additional_parameters: Optional[Dict] = None
+    additional_components: List[AdditionalComponent] = field(default_factory=list)
+    notes: Optional[MultilineHypertext] = None
+    # BOM options
+    add_up_in_bom: Optional[bool] = None
```

Comment:
Why defining the `add_up_in_bom` member? It seems not to be used at all.

Reply:
It is not used *yet*, but it stems from a wish to be able to specify how to handle lengths in the BOM.

Example:
A harness contains two identical cables of 1m each, another cable of same type but 5m, and two bundles of four identical wires of 1m each.
1. `add_up_in_bom = True` would show the BOM requiring 7m of cable + 8m of wire
  (that is the current behavior)
1. `add_up_in_bom = False` would show the BOM requiring 2 pcs. of cable (1m) + 1 pcs. of cable (5m) + 8 pcs. of wire (1m)

In a workflow where the person asssembling the harness is expecting to cut the cables themselves, option 1 is fine. If they are expecting to receive all cables/wires cut to length, option 2 is the way to go.

Another typical example is heatshrink: If there is a connector that requires 10mm of heatshrink on each of its 15 pins, the BOM could show either
1. 150mm of heatshrink or
2. 15 pcs. of 10mm heatshrink, which is much more understandable in context.

To avoid confusion while it is not implemented, feel free to comment the line out and add a `TODO` for later and I can write a new issue.

Reply:
True. See my [comment](https://github.com/wireviz/WireViz/pull/251#discussion_r1361674997) above for `sum_amounts_in_bom`. Seems this is a duplicate under a different name, and can go.